src/unix/linux_like/linux/arch/generic/mod.rs

```diff
@@ -204,14 +204,34 @@ pub const BLKIOOPT: ::Ioctl = 0x1279;
 pub const BLKSSZGET: ::Ioctl = 0x1268;
 pub const BLKPBSZGET: ::Ioctl = 0x127B;
 
-pub const FS_IOC_GETFLAGS: ::Ioctl = 1;
-pub const FS_IOC_SETFLAGS: ::Ioctl = 2;
-pub const FS_IOC_GETVERSION: ::Ioctl = 1;
-pub const FS_IOC_SETVERSION: ::Ioctl = 2;
-pub const FS_IOC32_GETFLAGS: ::Ioctl = 1;
-pub const FS_IOC32_SETFLAGS: ::Ioctl = 2;
-pub const FS_IOC32_GETVERSION: ::Ioctl = 1;
-pub const FS_IOC32_SETVERSION: ::Ioctl = 2;
+cfg_if! {
+    // Those type are constructed using the _IOC macro
+    // DD-SS_SSSS_SSSS_SSSS-TTTT_TTTT-NNNN_NNNN
+    // where D stands for direction (either None (00), Read (01) or Write (11))
+    // where S stands for size (int, long, struct...)
+    // where T stands for type ('f','v','X'...)
+    // where N stands for NR (NumbeR)
+    if #[cfg(any(target_arch = "x86"))] {
```

Comment:
This `any` doesn't do anything and should go to an arch-specific file.

Reply:
https://doc.rust-lang.org/reference/conditional-compilation.html

Reply:
Uhm I don't see an arch specific for x86 and x86_64 thus why I added them to generic, note sparc, powerpc and mips which are all the arch specific supported by libc for linux-like are covered so by doing so I think all case are covered.

The superfluous any were removed

Reply:
You could make a new module like `mips`. This structure is for arch-dependent declarations and using cfgs would lose the meaning here.